controllers/discordactions.js

```diff
@@ -272,6 +272,27 @@ const updateDiscordNicknames = async (req, res) => {
   }
 };
 
+/**
+ * Update all user Discord nickname based on status
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const updateUsersNicknameStatus = async (req, res) => {
```

Comment:
Please improve this code block on below points

1. Use a prepared statement: This will help to prevent SQL injection attacks.
2. Use a transaction: This will ensure that all of the changes are committed or rolled back together.
3. Use a batch update: This will improve performance if there are a large number of users to update.
4. Use a worker pool: This will improve performance if the code is being called concurrently by multiple threads.

Reply:
1. I am not sure I understand this. I thought the attacks occur only when SQL commands are built and instead of user input malicious SQL statements are provided. Can you please help me with this? Because when I searched on this in the context of firestore, I was unable to find anything about injection attacks. Also could not find ways to add prepared statements like in an SQL database.
2. Sure, I have made the changes to ensure that either all the nicknames are updated together or none of them are.
3. In the `models/discordactions.js`, I am updating the nickname in the batches of 4 Ensuring that there's a delay between each batch's nickname update.
4. For this I would require time to read and understand about worker pool. Please tell me if it's okay if I can take this as a separate extension and enhancement of this task.